Sources/PubNub/Helpers/WeakBox.swift

```diff
@@ -77,3 +77,15 @@ extension WeakSet: Collection {
     elements.lockedRead { $0.index(after: index) }
   }
 }
+
+extension WeakSet where Element == BaseSubscriptionListener {
+  func forEach(_ body: (BaseSubscriptionListener) throws -> Void) rethrows {
+    try elements.lockedTry {
+      try $0.compactMap {
+        $0.underlying
+      }.forEach {
+        try body($0)
+      }
+    }
+  }
+}
```

Comment:
Is it reasonable to make it only for a specific type of element? If we use it with another type, then it won't be thread safe. I mean, if we use type with thread-safety, why thread-safe version of `forEach` is done only for specific type?

Reply:
We could create a separate ticket for that, cause this requires more alteration of WeakSet than it's needed. Another thing is that migration to Swift 6 should help us dealing with threading issues in the future